interfaces/system_key_test.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package interfaces_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/osutil"
+)
+
+type systemKeySuite struct {
+	tmp              string
+	apparmorFeatures string
+	buildID          string
+}
+
+var _ = Suite(&systemKeySuite{})
+
+func (s *systemKeySuite) SetUpTest(c *C) {
+	s.tmp = c.MkDir()
+	dirs.SetRootDir(s.tmp)
+	s.apparmorFeatures = filepath.Join(s.tmp, "/sys/kernel/security/apparmor/features")
+
+	id, err := osutil.MyBuildID()
+	c.Assert(err, IsNil)
+	s.buildID = id
+}
+
+func (s *systemKeySuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+func (s *systemKeySuite) TestInterfaceSystemKeyEmpty(c *C) {
+	systemKey := interfaces.SystemKey()
+	c.Check(systemKey, Equals, fmt.Sprintf(`build-id: %s
+apparmor-features: []
+`, s.buildID))
+}
+
+func (s *systemKeySuite) TestInterfaceSystemKey(c *C) {
+	err := os.MkdirAll(s.apparmorFeatures, 0755)
+	c.Assert(err, IsNil)
+	err = os.MkdirAll(filepath.Join(s.apparmorFeatures, "policy"), 0755)
+	c.Assert(err, IsNil)
+
+	systemKey := interfaces.SystemKey()
+	c.Check(systemKey, Equals, fmt.Sprintf(`build-id: %s
+apparmor-features:
+- policy
+`, s.buildID))
+}
```

Comment:
Can we have a test for the case where apparmor is not available (in which case we get just the buildid, but no error is reported).

Reply:
Right now the system-key without apparmor would be:
```
build-id: 123
apparmor-features: []
```
and the responding test is called `TestInterfaceSystemKeyNoApparmor`. However if you prefer that if no apparmor features are available this is omited, thats easy too, just let me know.

Reply:
Of course, I'm sorry, I misread the first line of that test thinking it creates SystemKey stuct without ever exceuting apparmor probing code. That's perfect, thanks, empty apparmor-features array is fine.

Reply:
I think it would be best if it were `omitempty`, for symmetry with the seccomp features list which is also not there :-)